cordova-lib/src/plugman/fetch.js

```diff
@@ -70,6 +70,10 @@ function fetchPlugin(plugin_src, plugins_dir, options) {
             //if --fetch was used, throw error for subdirectories
             if (result[2] && options.fetch) {
                 return Q.reject(new CordovaError('--fetch does not support subdirectories'));
+            } else {
+                if(result[2]) {
+                    events.emit('warn', 'support for subdirectories is depreated and will be removed in Cordova@7');
+                }
```

Comment:
Yes, Sterling and I have resolved this and will push in the morning.
We did a few things:
- used options.subdir defined above instead of result[2] for clarity
- removed nested `else { if() { } }` in favor of `else if() { }`